Server crashed during queries on with the given partition

**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
```
[2021/06/24 08:46:41.855 +00:00] [WARN] [flow_graph_filter_dm_node.go:79] ["Non supporting"] ["message type"=200]
[2021/06/24 08:46:41.855 +00:00] [DEBUG] [query_collection.go:1066] ["get tSafe from flow graph"] [collectionID=425859324768419841] [tSafe=2021/06/24 08:46:41.785 +00:00]
[2021/06/24 08:46:41.856 +00:00] [DEBUG] [query_collection.go:281] ["not target collection retrieve request"] [collectionID=425859134687543297] [msgID=425858518492433365]
[2021/06/24 08:46:41.856 +00:00] [DEBUG] [query_collection.go:281] ["not target collection retrieve request"] [collectionID=425859135762333697] [msgID=425858518492436374]
[2021/06/24 08:46:41.856 +00:00] [DEBUG] [query_collection.go:281] ["not target collection retrieve request"] [collectionID=425859136850493441] [msgID=425858518492439383]
[2021/06/24 08:46:41.856 +00:00] [DEBUG] [query_collection.go:281] ["not target collection retrieve request"] [collectionID=425859227395031041] [msgID=425858518492445412]
[2021/06/24 08:46:41.856 +00:00] [DEBUG] [query_collection.go:281] ["not target collection retrieve request"] [collectionID=425859312513187841] [msgID=425858518492454439]
[2021/06/24 08:46:41.856 +00:00] [DEBUG] [query_collection.go:288] ["consume retrieve message"] [collectionID=425859324768419841] [msgID=425858518492454447]
[2021/06/24 08:46:41.856 +00:00] [DEBUG] [query_collection.go:333] ["doing retrieve in receiveRetrieveMsg..."] [collectionID=425859324768419841] [msgID=425858518492454447]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1503c7f]

goroutine 132356 [running]:
github.com/milvus-io/milvus/internal/querynode.(*queryCollection).retrieve(0xc001911800, 0xc0050608f0, 0x0, 0x0)
	/go/src/github.com/milvus-io/milvus/internal/querynode/query_collection.go:1246 +0xd5f
github.com/milvus-io/milvus/internal/querynode.(*queryCollection).receiveRetrieve(0xc001911800, 0xc0050608f0)
	/go/src/github.com/milvus-io/milvus/internal/querynode/query_collection.go:337 +0x1533
github.com/milvus-io/milvus/internal/querynode.(*queryCollection).consumeQuery(0xc001911800)
	/go/src/github.com/milvus-io/milvus/internal/querynode/query_collection.go:239 +0x275
created by github.com/milvus-io/milvus/internal/querynode.(*queryCollection).start
	/go/src/github.com/milvus-io/milvus/internal/querynode/query_collection.go:98 +0xa5

```
**Steps/Code to reproduce behavior**
```
entity/test_query.py::TestQueryPartition::test_query_empty_partition 
-------------------------------------------------------------------------------------- live log call --------------------------------------------------------------------------------------
[2021-06-24 16:46:41-ERROR-pymilvus.client.grpc_handler]: 
Addr [192.168.1.240:19530] query
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	debug_error_string = "{"created":"@1624524401.876485942","description":"Error received from peer ipv4:192.168.1.240:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Socket closed","grpc_status":14}"
>
	{'API start': '2021-06-24 16:46:41.821036', 'RPC start': '2021-06-24 16:46:41.821661', 'RPC error': '2021-06-24 16:46:41.877438'} (grpc_handler.py:72)
[2021-06-24 16:47:13-ERROR-pymilvus.client.grpc_handler]: Retry to connect server 192.168.1.240:19530 failed. (grpc_handler.py:338)
FAILED                                                                                                                                                                              [100%]
------------------------------------------------------------------------------------ live log teardown ------------------------------------------------------------------------------------
[2021-06-24 16:47:14-INFO-root]: connection was already closed! (conftest.py:126)

entity/test_query.py::TestQueryPartition::test_query_empty_partition ERROR                                                                                                          [100%]

========================================================================================= ERRORS ==========================================================================================
___________________________________________________________ ERROR at teardown of TestQueryPartition.test_query_empty_partition ____________________________________________________________

tp = <class 'AttributeError'>, value = None, tb = None

    def reraise(tp, value, tb=None):
        try:
            if value is None:
                value = tp()
            if value.__traceback__ is not tb:
                raise value.with_traceback(tb)
>           raise value

/home/zw/.local/lib/python3.6/site-packages/six.py:703: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/home/zw/.local/lib/python3.6/site-packages/six.py:703: in reraise
    raise value
/home/zw/.local/lib/python3.6/site-packages/six.py:703: in reraise
    raise value
conftest.py:179: in teardown
    if connect.has_collection(collection_name):
/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:63: in handler
    raise e
/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:51: in handler
    return func(self, *args, **kwargs)
/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:40: in inner
    return func(self, *args, **kwargs)
/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:278: in has_collection
    with self._connection() as handler:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pymilvus.client.stub.Milvus object at 0x7f8bcae18eb8>

    def _connection(self):
>       return self._pool.fetch()
E       AttributeError: 'NoneType' object has no attribute 'fetch'

/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:162: AttributeError
------------------------------------------------------------------------------------ Captured log call ------------------------------------------------------------------------------------
[2021-06-24 16:46:41-ERROR-pymilvus.client.grpc_handler]: 
Addr [192.168.1.240:19530] query
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	debug_error_string = "{"created":"@1624524401.876485942","description":"Error received from peer ipv4:192.168.1.240:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Socket closed","grpc_status":14}"
>
	{'API start': '2021-06-24 16:46:41.821036', 'RPC start': '2021-06-24 16:46:41.821661', 'RPC error': '2021-06-24 16:46:41.877438'} (grpc_handler.py:72)
[2021-06-24 16:47:13-ERROR-pymilvus.client.grpc_handler]: Retry to connect server 192.168.1.240:19530 failed. (grpc_handler.py:338)
---------------------------------------------------------------------------------- Captured log teardown ----------------------------------------------------------------------------------
[2021-06-24 16:47:14-INFO-root]: connection was already closed! (conftest.py:126)
======================================================================================== FAILURES =========================================================================================
______________________________________________________________________ TestQueryPartition.test_query_empty_partition ______________________________________________________________________

self = <pymilvus.client.stub.Milvus object at 0x7f8bcae18eb8>, args = ('test_JZ5zjP8Y', 'int64 in [0, 1, 2, 3, 4]'), kwargs = {'partition_names': ['_default']}, counter = 2

    @functools.wraps(func)
    def handler(self, *args, **kwargs):
        counter = 1
        try:
>           return func(self, *args, **kwargs)

/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pymilvus.client.stub.Milvus object at 0x7f8bcae18eb8>, args = ('test_JZ5zjP8Y', 'int64 in [0, 1, 2, 3, 4]'), kwargs = {'partition_names': ['_default']}

    @functools.wraps(func)
    def inner(self, *args, **kwargs):
>       return func(self, *args, **kwargs)

/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pymilvus.client.stub.Milvus object at 0x7f8bcae18eb8>, collection_name = 'test_JZ5zjP8Y', expr = 'int64 in [0, 1, 2, 3, 4]', output_fields = None, partition_names = ['_default']
timeout = None

    @retry_on_rpc_failure(retry_times=10, wait=1)
    @check_connect
    def query(self, collection_name, expr, output_fields=None, partition_names=None, timeout=None):
        """
        Query with a set of criteria, and results in a list of records that match the query exactly.
    
        :param collection_name: Name of the collection to retrieve entities from
        :type  collection_name: str
    
        :param expr: The query expression
        :type  expr: str
    
        :param output_fields: A list of fields to return
        :type  output_fields: list[str]
    
        :param partition_names: Name of partitions that contain entities
        :type  partition_names: list[str]
    
        :param timeout: An optional duration of time in seconds to allow for the RPC. When timeout
                        is set to None, client waits until server response or error occur
        :type  timeout: float
    
        :return: A list that contains all results
        :rtype: list
    
        :raises:
            RpcError: If gRPC encounter an error
            ParamError: If parameters are invalid
            BaseException: If the return result from server is not ok
        """
        with self._connection() as handler:
>           return handler.query(collection_name, expr, output_fields, partition_names, timeout=timeout)

/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:1117: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pymilvus.client.grpc_handler.GrpcHandler object at 0x7f8bcae185f8>, args = ('test_JZ5zjP8Y', 'int64 in [0, 1, 2, 3, 4]', None, ['_default']), kwargs = {'timeout': None}
record_dict = {'API start': '2021-06-24 16:46:41.821036', 'RPC error': '2021-06-24 16:46:41.877438', 'RPC start': '2021-06-24 16:46:41.821661'}

    def handler(self, *args, **kwargs):
        record_dict = {}
        try:
            record_dict["API start"] = str(datetime.datetime.now())
            if self._pre_ping:
                self.ping()
            record_dict["RPC start"] = str(datetime.datetime.now())
            return func(self, *args, **kwargs)
        except BaseException as e:
            LOGGER.error("Error: {}".format(e))
            if e.code == Status.ILLEGAL_COLLECTION_NAME:
                raise IllegalCollectionNameException(e.code, e.message)
            if e.code == Status.COLLECTION_NOT_EXISTS:
                raise CollectionNotExistException(e.code, e.message)
    
            raise e
        except grpc.FutureTimeoutError as e:
            record_dict["RPC timeout"] = str(datetime.datetime.now())
            LOGGER.error("\nAddr [{}] {}\nRequest timeout: {}\n\t{}".format(self.server_address, func.__name__, e,
                                                                            record_dict))
            raise e
        except grpc.RpcError as e:
            record_dict["RPC error"] = str(datetime.datetime.now())
            LOGGER.error(
                "\nAddr [{}] {}\nRPC error: {}\n\t{}".format(self.server_address, func.__name__, e, record_dict))
>           raise e

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py:73: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pymilvus.client.grpc_handler.GrpcHandler object at 0x7f8bcae185f8>, args = ('test_JZ5zjP8Y', 'int64 in [0, 1, 2, 3, 4]', None, ['_default']), kwargs = {'timeout': None}
record_dict = {'API start': '2021-06-24 16:46:41.821036', 'RPC error': '2021-06-24 16:46:41.877438', 'RPC start': '2021-06-24 16:46:41.821661'}

    def handler(self, *args, **kwargs):
        record_dict = {}
        try:
            record_dict["API start"] = str(datetime.datetime.now())
            if self._pre_ping:
                self.ping()
            record_dict["RPC start"] = str(datetime.datetime.now())
>           return func(self, *args, **kwargs)

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pymilvus.client.grpc_handler.GrpcHandler object at 0x7f8bcae185f8>, collection_name = 'test_JZ5zjP8Y', expr = 'int64 in [0, 1, 2, 3, 4]', output_fields = None
partition_names = ['_default'], timeout = None

    @error_handler()
    def query(self, collection_name, expr, output_fields=None, partition_names=None, timeout=None):
        request = Prepare.query_request(collection_name, expr, output_fields, partition_names)
        future = self._stub.Query.future(request, wait_for_ready=True, timeout=timeout)
>       response = future.result()

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py:1011: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	de...1.240:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Socket closed","grpc_status":14}"
>
timeout = None

    def result(self, timeout=None):
        """Returns the result of the computation or raises its exception.
    
        See grpc.Future.result for the full API contract.
        """
        with self._state.condition:
            timed_out = _common.wait(self._state.condition.wait,
                                     self._is_complete,
                                     timeout=timeout)
            if timed_out:
                raise grpc.FutureTimeoutError()
            else:
                if self._state.code is grpc.StatusCode.OK:
                    return self._state.response
                elif self._state.cancelled:
                    raise grpc.FutureCancelledError()
                else:
>                   raise self
E                   grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
E                   	status = StatusCode.UNAVAILABLE
E                   	details = "Socket closed"
E                   	debug_error_string = "{"created":"@1624524401.876485942","description":"Error received from peer ipv4:192.168.1.240:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Socket closed","grpc_status":14}"
E                   >

/usr/local/lib/python3.6/dist-packages/grpc/_channel.py:722: _MultiThreadedRendezvous

During handling of the above exception, another exception occurred:

self = <pymilvus.client.grpc_handler.GrpcHandler object at 0x7f8bcade4048>

    def ping(self):
        begin_timeout = 1
        timeout = begin_timeout
        ft = grpc.channel_ready_future(self._channel)
        retry = self._max_retry
        try:
            while retry > 0:
                try:
>                   ft.result(timeout=timeout)

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py:329: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <grpc._utilities._ChannelReadyFuture object at 0x7f8bcadf9160>, timeout = 16

    def result(self, timeout=None):
>       self._block(timeout)

/usr/local/lib/python3.6/dist-packages/grpc/_utilities.py:140: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <grpc._utilities._ChannelReadyFuture object at 0x7f8bcadf9160>, timeout = 16

    def _block(self, timeout):
        until = None if timeout is None else time.time() + timeout
        with self._condition:
            while True:
                if self._cancelled:
                    raise grpc.FutureCancelledError()
                elif self._matured:
                    return
                else:
                    if until is None:
                        self._condition.wait()
                    else:
                        remaining = until - time.time()
                        if remaining < 0:
>                           raise grpc.FutureTimeoutError()
E                           grpc.FutureTimeoutError

/usr/local/lib/python3.6/dist-packages/grpc/_utilities.py:86: FutureTimeoutError

During handling of the above exception, another exception occurred:

self = <test_query.TestQueryPartition object at 0x7f8bcae1e6d8>, connect = <pymilvus.client.stub.Milvus object at 0x7f8bcae18eb8>, collection = 'test_JZ5zjP8Y'

    def test_query_empty_partition(self, connect, collection):
        """
        target: test query on empty partition
        method: query on a empty collection
        expected: empty query result
        """
        connect.create_partition(collection, ut.default_tag)
        connect.load_partitions(collection, [ut.default_partition_name])
>       res = connect.query(collection, default_term_expr, partition_names=[ut.default_partition_name])

entity/test_query.py:537: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:61: in handler
    self._update_connection_pool()
/usr/local/lib/python3.6/dist-packages/pymilvus/client/stub.py:169: in _update_connection_pool
    self._pool = SingletonThreadPool(self._pool_uri, **self._pool_kwargs)
/usr/local/lib/python3.6/dist-packages/pymilvus/client/pool.py:231: in __init__
    self._prepare()
/usr/local/lib/python3.6/dist-packages/pymilvus/client/pool.py:236: in _prepare
    conn.client().ping()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pymilvus.client.grpc_handler.GrpcHandler object at 0x7f8bcade4048>

    def ping(self):
        begin_timeout = 1
        timeout = begin_timeout
        ft = grpc.channel_ready_future(self._channel)
        retry = self._max_retry
        try:
            while retry > 0:
                try:
                    ft.result(timeout=timeout)
                    return True
                except:
                    retry -= 1
                    LOGGER.debug("Retry connect addr <{}> {} times".format(self._uri, self._max_retry - retry))
                    if retry > 0:
                        timeout *= 2
                        continue
                    else:
                        LOGGER.error("Retry to connect server {} failed.".format(self._uri))
                        raise
        except grpc.FutureTimeoutError:
>           raise NotConnectError('Fail connecting to server on {}. Timeout'.format(self._uri))
E           pymilvus.client.exceptions.NotConnectError: Fail connecting to server on 192.168.1.240:19530. Timeout

/usr/local/lib/python3.6/dist-packages/pymilvus/client/grpc_handler.py:341: NotConnectError
------------------------------------------------------------------------------------ Captured log call ------------------------------------------------------------------------------------
[2021-06-24 16:46:41-ERROR-pymilvus.client.grpc_handler]: 
Addr [192.168.1.240:19530] query
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Socket closed"
	debug_error_string = "{"created":"@1624524401.876485942","description":"Error received from peer ipv4:192.168.1.240:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Socket closed","grpc_status":14}"
>
	{'API start': '2021-06-24 16:46:41.821036', 'RPC start': '2021-06-24 16:46:41.821661', 'RPC error': '2021-06-24 16:46:41.877438'} (grpc_handler.py:72)
[2021-06-24 16:47:13-ERROR-pymilvus.client.grpc_handler]: Retry to connect server 192.168.1.240:19530 failed. (grpc_handler.py:338)

```
**Expected behavior**
A clear and concise description of what you expected to happen.

**Method of installation**
- [ ] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Software conditions (OS, CPU, GPU, Memory)


- Milvus version (master or released version)


**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

--- a/internal/querynode/query_collection.go
+++ b/internal/querynode/query_collection.go
@@ -1286,6 +1286,14 @@ func mergeRetrieveResults(dataArr []*segcorepb.RetrieveResults) (*segcorepb.Retr
 		}
 	}
 
+	// not found, return default values indicating not result found
+	if final == nil {
+		final = &segcorepb.RetrieveResults{
+			Ids:        nil,
+			FieldsData: []*schemapb.FieldData{},
+		}
+	}
+
 	return final, nil
 }
 

